component/registry/registry.go

```diff
@@ -0,0 +1,63 @@
+package registry
+
+import (
+	"fmt"
+
+	"github.com/docker/docker/component"
+)
+
+// Registry stores Components so they can retrieved by other Components
+type Registry struct {
+	components map[string]component.Component
+}
+
+// Register adds a new component to the registry. If there is already a
+// component with the same ComponentType an error is returned.
+func (r *Registry) Register(c component.Component) {
+	r.components[c.Provides()] = c
```

Comment:
I know this is core stuff, but do we think there will ever be a time we'll want more than one of each of these components?